.github/workflows/release.yml

```diff
@@ -34,44 +45,46 @@ jobs:
           secrets: |
             totp/code/npmjs-elasticmachine code | TOTP_CODE
 
-      - uses: elastic/apm-pipeline-library/.github/actions/setup-npmrc@current
+      - name: Configure npm registry
+        uses: elastic/apm-pipeline-library/.github/actions/setup-npmrc@current
+        if: inputs.dry-run != null && !inputs.dry-run
         with:
           vault-url: ${{ secrets.VAULT_ADDR }}
           vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
           vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
           secret: secret/jenkins-ci/npmjs/elasticmachine
           secret-key: token
 
+      - name: Configure git user
+        uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+
       - name: Publish the release
+        env:
+          DRY_RUN: ${{ inputs.dry-run }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: npm run ci:release
 
-  release-cdn:
-    runs-on: ubuntu-latest
-    env:
-      BUCKET_NAME: 'apm-rum-357700bc'
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-
-      - uses: actions/setup-node@v3
-        with:
-          node-version-file: '.nvmrc'
-
-      - name: Install dependencies
-        run: npm ci
-
-      - name: Build dist
-        run: npm run build
-
-      - uses: hashicorp/vault-action@v2.5.0
+      - name: Setup credentials
+        env:
+          DRY_RUN: ${{ inputs.dry-run }}
+        run: |
+          if [ "${DRY_RUN}" == "false" ]; then
+            echo 'ELASTIC_CDN_BUCKET_NAME=apm-rum-357700bc' >> ${GITHUB_ENV}
+            echo 'ELASTIC_CDN_CREDENTIALS=secret/gce/elastic-cdn/service-account/apm-rum-admin' >> ${GITHUB_ENV}
+          else
+            echo 'ELASTIC_CDN_BUCKET_NAME=oblt-apm-agent-rum-js-ci' >> ${GITHUB_ENV}
+            echo 'ELASTIC_CDN_CREDENTIALS=secret/observability-team/ci/service-account/apm-agent-rum-js' >> ${GITHUB_ENV}
+          fi
```

Comment:
I am not sure of the goal of this step, if DRY_RUN=true all the rest of the steps that use those env vars will fail. The env vars can be defined at the job level in a `env` block.

Reply:
The `DRY_RUN` env var is used to define which secrets we will use.
We could use either the production secrets or the CI secrets.
Those secrets contain similar values but refer to different accounts.